helpers/calendar/analytics_helper.py

```diff
@@ -172,16 +173,35 @@ def get_room_statistics(self, number_of_events_in_room, all_details):
         return result
 
     @staticmethod
-    def get_total_bookings(self, query, start_date, end_date):
+    def get_total_bookings(instance, query, start_date, end_date, id=None):
         bookings = 0
-        rooms = CommonAnalytics.get_calendar_id_name(self, query)
+        rooms = CommonAnalytics.get_calendar_id_name(instance, query)
+        active_room = RoomModel.query.filter(RoomModel.state == "active")
+        exact_room = active_room.filter(RoomModel.id == id).first()
+        if id and not exact_room:
```

Comment:
Why do you check for id at this point yet you have used it, same to in line 185 and 190?

Reply:
@Bonifase since we need to be able to return bookings for when an id is provided and when it isn't.
when an id is passed, then only the booking count for that particular room is returned whereas when no id is passed, then the total booking count for all the rooms in a particular location is returned.